internal/servers/worker/internal/metric/proxy_http.go

```diff
@@ -0,0 +1,105 @@
+// Package metric provides functions to initialize the worker specific
+// collectors and hooks to measure metrics and update the relevant collectors.
+package metric
+
+import (
+	"fmt"
+	"net/http"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/boundary/globals"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const (
+	invalidPathValue = "invalid"
+	proxyPathValue   = "/v1/proxy"
+
+	labelHttpCode   = "code"
+	labelHttpPath   = "path"
+	labelHttpMethod = "method"
+	proxySubSystem  = "worker_proxy"
+)
+
+// httpTimeUntilHeader collects measurements of how long it takes
+// the boundary system to hijack an HTTP request into a websocket connection
+// for the proxy worker.
+var httpTimeUntilHeader prometheus.ObserverVec = prometheus.NewHistogramVec(
+	prometheus.HistogramOpts{
+		Namespace: globals.MetricNamespace,
+		Subsystem: proxySubSystem,
+		Name:      "http_write_header_duration_seconds",
+		Help:      "Histogram of latencies for HTTP to websocket conversions.",
```

Comment:
This help text isn't accurate any more.  It should state that this is a histogram of the latency from the time the request is received (post TLS handshake) to when the first http header is written back from the server.